### PR TITLE
use preg_replace instead of mb_strcut

### PR DESCRIPTION
**Pull request by gharlan**
thanks to @bwoebi for the regex (I had to adjust it to add a space before each line).

In my case it is much faster:

screenshot 2018-04-18 12 59 56

### CODE REVIEW

**Comment by gharlan** [comment]
> should/can we use the same optimization in

Yes, but it is less important, because the part is only used for VCard 2.1
https://github.com/sabre-io/vobject/blob/38d1d014ae6f909de45b9bc8cf3bb6ca1fa55db3/lib/Property/Text.php#L209-L212
(current version is VCard 4.0)

**Comment by staabm** [comment]
nevertheless, please keep the code in sync.

File: lib/Property.php [hunk 1]
```diff
@@ -246,20 +246,9 @@ function serialize() {
...
-                $part = mb_strcut($str, 0, 75, 'utf-8');
-                $out .= $part . "\r\n";
-                $str = ' ' . substr($str, strlen($part));
-            } else {
-                $out .= $str . "\r\n";
-                $str = '';
-                break;
-            }
-        }
+        $str = \preg_replace('/((?:^.)?.{1,74}(?![\x80-\xbf]))/', "$1\r\n ", $str);
```
Comment (staabm) [inline lib/Property.php]: did you intenionally add a space after `\r\n` ?

**Comment by evert** [comment]
Have you considered using the /x syntax and add newlines and comments to make it more readable?

There's really no need for these hard to parse oneliners in PHP

**Comment by evert** [comment]
Does this work for lines that need multiple breaks? I'm also a bit worried about the lack of a /u flag. Doesn't that mean the regex operates on bytes instead of characters?

**Comment by gharlan** [comment]
> Have you considered using the /x syntax and add newlines and comments to make it more readable?

I will do that.

> Does this work for lines that need multiple breaks?

I will add some more tests.

> Doesn't that mean the regex operates on bytes instead of characters?

Yeah, we need to cut after 75 bytes (like before by using `mb_strcut` which also operates on bytes).

**Comment by evert** [comment]
One thing mb_strcut does though is that it never splits a multibyte character in 2.

**Comment by bwoebi** [comment]
@evert That's what `(?![\x80-\xbf])` does - it requires the next character to be ASCII or introducing an unicode codepoint (i.e. it must not be in the middle of a character).

**Comment by gharlan** [comment]
`(?![\x80-\xbf])` should prevent splitting multibyte characters. 
The regex is from here: https://chat.stackoverflow.com/transcript/message/42160271#42160271
I hope that `(?![\x80-\xbf])` is 100% the same as with `mb_strcut`.

**Comment by staabm** [comment]
Please resolve conflicts

**Comment by gharlan** [comment]
> Please resolve conflicts

done

File: tests/VObject/PropertyTest.php [hunk 2]
```diff
@@ -161,9 +161,14 @@ function testSerializeLongLine() {
     function testSerializeUTF8LineFold() {
 
         $cal = new VCalendar();
-        $value = str_repeat('!', 65) . "\xc3\xa4bla"; // inserted umlaut-a
+        $value = str_repeat('!', 65) . "\xc3\xa4bla" . str_repeat('!', 142) . "\xc3\xa4foo"; // inserted umlaut-a
```
[the remark this answers is not in the document]
Comment (gharlan) [reply tests/VObject/PropertyTest.php]: No, I just made the value longer to test some more line breaks. But the the first line break is still expectected at the same position.

**Comment by staabm** [comment]
thx
